apps/src/aichat/types.ts

```diff
@@ -146,3 +146,37 @@ export interface LevelAichatSettings {
 
 // The type of save action being performed (customization update, publish, model card save, etc).
 export type SaveType = 'updateChatbot' | 'publishModelCard' | 'saveModelCard';
+
+/** Response structure for the detect toxicity API */
+export interface DetectToxicityResponse {
+  flaggedFields: FlaggedField[];
+}
+
+export interface FlaggedField {
+  field: keyof AiCustomizations;
+  toxicity: {
+    text: string;
+    blockedBy: SafetyService;
+    details: BlocklistDetails | WebPurifyDetails | ComprehendDetails;
+  };
+}
+
+type SafetyService = 'blocklist' | 'webpurify' | 'comprehend';
+
+interface BlocklistDetails {
+  blockedWord: string;
+}
+
+interface WebPurifyDetails {
+  type: string;
```

Comment:
Can we set as either `profanity`, `email`, `address`, `phone`? https://github.com/code-dot-org/code-dot-org/blob/2051613523466f477eae425e8458137ad53ffbea/lib/cdo/share_filtering.rb#L38-L43

Reply:
Done!